jumpscale/packages/tfgrid_solutions/actors/solutions.py

```diff
@@ -110,6 +115,10 @@ def update_pool_farm_cache(cache_dict, pool, farm):
 
     @actor_method
     def cancel_workload(self, wid) -> bool:
+        workload = j.sals.zos.get().workloads.get(wid)
+        # Delete public ip
+        if workload.public_ip:
+            j.sals.zos.get().workloads.decomission(workload.public_ip)
         j.sals.zos.get().workloads.decomission(wid)
```

Comment:
getting a zos object each time (around 1 second), takes time, what about we get it once store it in a variable and use it

Reply:
Done 
commit 
https://github.com/threefoldtech/js-sdk/pull/2447/commits/bc82d7ad248864190be95995bff4195234d4742e